GHermeneus/include/GHermeneus/GCode.h

```diff
@@ -0,0 +1,103 @@
+//
+// Created by Jelle Spijker on 7/31/20.
+//
+
+#ifndef GCODEHERMENEUS_GCODE_H
+#define GCODEHERMENEUS_GCODE_H
+
+#include <unordered_map>
+#include <string_view>
+#include <functional>
+
+#include "Parameters.h"
+
+namespace GHermeneus
+{
+    /*!
+     * @brief An extracted GCode line containing the line_nr and the GCode line as a string_view
+     */
+    using Line = std::pair<size_t, std::string_view>;
+
+    /*!
+     * @brief The Function type for where the conversion from specific GCode command takes place, which while transform
+     * a previous State Space Function, given as parameter, to a new parameter. These are dialect specific.
+     * @tparam SSV_T State Space Vector Type of the dialect
+     * @tparam T primative type used in the State Space Vector and the parameters
+     */
+    template<typename SSV_T, typename T>
+    using GCodeFunction = std::function<SSV_T(const SSV_T&, const Parameters<T>&)>;
+
+    /*!
+     * @brief An Unorded map with a lookup key for a specific dialect GCode command (e.q. G0, T1, M104) and a GCodeFucntion
+     * @tparam GCFUNC_T the GCodeFunction type
+     */
+    template<typename GCFUNC_T>
+    using CmdMap = std::unordered_map<std::string_view, GCFUNC_T>;
+
+    /*!
+     * Todo: this current format will probably not be used
```

Comment:
Probably not that important in the current state, but just a reminder for the future that we at least like to keep up the pretence of not leaving TODO's in code in master ;-)

(I do so now in the libArachne branch, because all those todo's are things that should at least be looked at _before_ we merge.)

Reply:
Totally agree with you. I normally also leave them out. But because the project as a whole is still very much a WIP and subject to big changes I left them in as reminders.
They will however be removed at a later stage.